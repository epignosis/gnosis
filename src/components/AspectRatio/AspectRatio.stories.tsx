import React from "react";
import { Story } from "@storybook/react";
import AspectRatio, { AspectRatioProps } from "./AspectRatio";

export default {
  title: "Components/AspectRatio",
  component: AspectRatio,
};

export const Default: Story<AspectRatioProps> = (args) => (
  <AspectRatio {...args}>
    <img
      src="https://d3j0t7vrtr92dk.cloudfront.net/test/1616506869_Def_Course.png?"
      alt="An image"
      height="100%"
      width="100%"
    />
  </AspectRatio>
);

Default.args = {
  ratio: [4, 3],
};

Default.parameters = {
  viewport: {
    defaultViewport: "xs",
  },
};
