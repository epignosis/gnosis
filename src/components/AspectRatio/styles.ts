import { css, SerializedStyles } from "@emotion/react";

export const aspectRatioContainer = (padding: number): SerializedStyles => {
  return css`
    position: relative;
    height: 0;
    padding-top: ${padding}%;
    overflow: hidden;

    .aspect-ratio-wrapper {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
    }
  `;
};
