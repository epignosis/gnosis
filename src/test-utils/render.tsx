import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
