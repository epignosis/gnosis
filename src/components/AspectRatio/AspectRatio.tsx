import React, { FC } from "react";
import { aspectRatioContainer } from "./styles";

export type AspectRatioProps = {
  ratio: [number, number];
};

const AspectRatio: FC<AspectRatioProps> = ({ ratio, children }) => {
  const paddingPercentage = (ratio[1] / ratio[0]) * 100;

  return (
    <div css={aspectRatioContainer(paddingPercentage)}>
      <div className="aspect-ratio-wrapper">{children}</div>
    </div>
  );
};

export default AspectRatio;
