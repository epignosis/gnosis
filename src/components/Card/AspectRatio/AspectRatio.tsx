import React, { FC } from "react";
import { aspectRatioContainer } from "./styles";

export type AspectRatioProps = React.HTMLAttributes<HTMLDivElement> & {
  ratio: [number, number];
};

const AspectRatio: FC<AspectRatioProps> = ({ ratio, children, ...rest }) => {
  const paddingPercentage = (ratio[1] / ratio[0]) * 100;

  return (
    <div css={aspectRatioContainer(paddingPercentage)} {...rest}>
      <div className="aspect-ratio-wrapper">{children}</div>
    </div>
  );
};

export default AspectRatio;
