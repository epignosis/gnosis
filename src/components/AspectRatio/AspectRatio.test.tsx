import React from "react";
import faker from "faker";
import AspectRatio from "./AspectRatio";
import { render, screen } from "@test-utils/render";

describe("<AspectRatio />", () => {
  it("renders correctly", () => {
    const alt = faker.lorem.word();
    const src = faker.image.imageUrl();

    render(
      <AspectRatio ratio={[4, 3]}>
        <img src={src} alt={alt} height="100%" width="100%" />
      </AspectRatio>,
    );
    const img = screen.getByAltText(alt);

    expect(img).toHaveAttribute("src", src);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <AspectRatio ratio={[4, 3]}>
        <img
          src="https://d3j0t7vrtr92dk.cloudfront.net/test/1616506869_Def_Course.png"
          alt="An image"
          height="100%"
          width="100%"
        />
      </AspectRatio>,
    );

    expect(container).toMatchSnapshot();
  });
});
