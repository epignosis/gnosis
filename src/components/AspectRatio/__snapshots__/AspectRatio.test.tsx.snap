// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AspectRatio /> matches snapshot 1`] = `
<div>
  <div
    class="my-class css-14dee1b-aspectRatioContainer-aspectRatioContainer"
    id="my-id"
  >
    <div
      class="aspect-ratio-wrapper"
    >
      <img
        alt="An image"
        height="100%"
        src="https://d3j0t7vrtr92dk.cloudfront.net/test/1616506869_Def_Course.png"
        width="100%"
      />
    </div>
  </div>
</div>
`;
